sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -415,11 +424,10 @@ def typed_metric(self, cfg_inst, table, base_name=None, user_type=None, sql_type
 
     def check(self, _):
         if self.do_check:
-            proc = self.instance.get('stored_procedure')
-            if proc is None:
-                self.collect_metrics()
+            if self.proc:
+                self.do_stored_procedure_check()
```

Comment:
nit: should we log this with a message so that users know only their stored_procedures are run? I know this is existing behavior but good for debugging imo

Reply:
Not sure we need to, its the current behavior and we document it in the conf.yaml file